<launch>

    <!-- Configurable arguments -->
    <arg name="robot_name" default="sam"/>

    <arg name="loop_freq" default="20."/>
    <arg name="verbose_arg" default="True"/>

    <arg name="waypoint_topic_arg" default="/waypoint_topic"/>


    <!-- topics, note the root! -->
    <arg name="state_estimation_topic" default="/$(arg robot_name)/slam/odom" />
    <arg name="ref_pose_topic_arg" default="$(arg waypoint_topic_arg)" />

    <arg name="vbs_topic_arg" default="/$(arg robot_name)/core/vbs_cmd" />
    <arg name="lcg_topic_arg" default="/$(arg robot_name)/core/lcg_cmd" />
    <arg name="rpm_topic_1_arg" default="/$(arg robot_name)/core/thruster1_cmd" />
    <arg name="rpm_topic_2_arg" default="/$(arg robot_name)/core/thruster2_cmd" />
    <arg name="thrust_vector_cmd_topic_arg" default="/$(arg robot_name)/core/thrust_vector_cmd" />


    <!-- Nodes -->
    <node name="WaypointFollowingController" pkg="visual_feedback" type="WaypointFollowingController.py"
        output="screen" ns="$(arg robot_name)/ctrl">

        <param name="loop_freq" value="$(arg loop_freq)"/>
        <param name="verbose" value="$(arg verbose_arg)"/>

        <!-- Subscriber State Topics  -->
        <param name="ref_pose_topic" value="$(arg ref_pose_topic_arg)" />
        <param name="state_estimation_topic" value="$(arg state_estimation_topic)" />

        <!-- Publisher Actuator Topics -->
        <param name="vbs_topic" value="$(arg vbs_topic_arg)" />
        <param name="lcg_topic" value="$(arg lcg_topic_arg)" />
        <param name="rpm_topic_1" value="$(arg rpm_topic_1_arg)" />
        <param name="rpm_topic_2" value="$(arg rpm_topic_2_arg)" />
        <param name="thrust_vector_cmd_topic" value="$(arg thrust_vector_cmd_topic_arg)" />

        <!-- Publish Convenience Topics -->
        <param name="ref_topic" value="/$(arg robot_name)/ctrl/setpoint" />
        <param name="error_topic" value="/$(arg robot_name)/ctrl/error" />
        <param name="state_topic" value="/$(arg robot_name)/ctrl/state" />
        <param name="control_input_topic" value="/$(arg robot_name)/ctrl/control_input" />
        <param name="control_neutral_topic" value="/$(arg robot_name)/ctrl/control_neutral" />
    </node>

</launch>
